### PR TITLE
domain,session: fix annoying log when running session bench test

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

```
cd session
go test -check.f XXX -test.bench Benchmark
```

What I want:

```
BenchmarkBasic-16                              	   38066	     28349 ns/op
BenchmarkTableScan-16                          	    6150	    205441 ns/op
BenchmarkExplainTableScan-16                   	   19684	     54121 ns/op
BenchmarkTableLookup-16                        	   16795	     74477 ns/op
BenchmarkExplainTableLookup-16                 	   43236	     31524 ns/op
BenchmarkStringIndexScan-16                    	    4254	    280159 ns/op
BenchmarkExplainStringIndexScan-16             	   14622	     81911 ns/op
BenchmarkStringIndexLookup-16                  	    6004	    195449 ns/op
BenchmarkIntegerIndexScan-16                   	    4393	    278203 ns/op
BenchmarkIntegerIndexLookup-16                 	    6073	    205663 ns/op
BenchmarkDecimalIndexScan-16                   	    3489	    328341 ns/op
BenchmarkDecimalIndexLookup-16                 	    5578	    209139 ns/op
BenchmarkInsertWithIndex-16                    	    4204	    263469 ns/op
BenchmarkInsertNoIndex-16                      	    4508	    254953 ns/op
BenchmarkSort-16                               	     218	   5502274 ns/op
BenchmarkJoin-16                               	    1786	    633263 ns/op
BenchmarkJoinLimit-16                          	    1785	    679006 ns/op
BenchmarkPartitionPruning-16                   	     778	   1482036 ns/op
BenchmarkRangeColumnPartitionPruning-16        	f    1597	    752830 ns/op
BenchmarkHashPartitionPruningPointSelect-16    	    2348	    466922 ns/op
BenchmarkHashPartitionPruningMultiSelect-16    	     781	   1582944 ns/op
PASS
ok  	github.com/pingcap/tidb/session	85.420s
```
What I get:

```
BenchmarkBasic-16                              	   36102	     28502 ns/op
[2021/06/28 15:05:36.203 +08:00] [ERROR] [handle.go:683] ["[stats] dump stats delta fail"] [error="context canceled"] [errorVerbose="context canceled\ngithub.com/pingcap/errors.AddStack\n\t/home/genius/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.Trace\n\t/home/genius/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/juju_adaptor.go:15\ngithub.com/tikv/client-go/v2/locate.(*RegionRequestSender).onSendFail\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/locate/region_request.go:789\ngithub.com/tikv/client-go/v2/locate.(*RegionRequestSender).sendReqToRegion\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/locate/region_request.go:751\ngithub.com/tikv/client-go/v2/locate.(*RegionRequestSender).SendReqCtx\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/locate/region_request.go:529\ngithub.com/tikv/client-go/v2/tikv.(*ClientHelper).SendReqCtx\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/tikv/client_helper.go:83\ngithub.com/tikv/client-go/v2/tikv.(*KVSnapshot).get\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/tikv/snapshot.go:521\ngithub.com/tikv/client-go/v2/tikv.(*KVSnapshot).Get\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/tikv/snapshot.go:444\ngithub.com/pingcap/tidb/store/driver/txn.(*tikvSnapshot).Get\n\t/home/genius/project/src/github.com/pingcap/tidb/store/driver/txn/snapshot.go:45\ngithub.com/pingcap/tidb/executor.(*PointGetExecutor).get\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/point_get.go:405\ngithub.com/pingcap/tidb/executor.(*PointGetExecutor).Next\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/point_get.go:236\ngithub.com/pingcap/tidb/executor.Next\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/executor.go:286\ngithub.com/pingcap/tidb/executor.(*UpdateExec).updateRows\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/update.go:251\ngithub.com/pingcap/tidb/executor.(*UpdateExec).Next\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/update.go:219\ngithub.com/pingcap/tidb/executor.Next\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/executor.go:286\ngithub.com/pingcap/tidb/executor.(*ExecStmt).handleNoDelayExecutor\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/adapter.go:589\ngithub.com/pingcap/tidb/executor.(*ExecStmt).handleNoDelay\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/adapter.go:470\ngithub.com/pingcap/tidb/executor.(*ExecStmt).Exec\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/adapter.go:419\ngithub.com/pingcap/tidb/session.runStmt\n\t/home/genius/project/src/github.com/pingcap/tidb/session/session.go:1635\ngithub.com/pingcap/tidb/session.(*session).ExecuteStmt\n\t/home/genius/project/src/github.com/pingcap/tidb/session/session.go:1530\ngithub.com/pingcap/tidb/session.(*session).ExecuteInternal\n\t/home/genius/project/src/github.com/pingcap/tidb/session/session.go:1250\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).dumpTableStatCountToKV.func2\n\t/home/genius/project/src/github.com/pingcap/tidb/statistics/handle/update.go:473\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).dumpTableStatCountToKV\n\t/home/genius/project/src/github.com/pingcap/tidb/statistics/handle/update.go:477\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).DumpStatsDeltaToKV\n\t/home/genius/project/src/github.com/pingcap/tidb/statistics/handle/update.go:425\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).FlushStats\n\t/home/genius/project/src/github.com/pingcap/tidb/statistics/handle/handle.go:682\ngithub.com/pingcap/tidb/domain.(*Domain).updateStatsWorker\n\t/home/genius/project/src/github.com/pingcap/tidb/domain/domain.go:1279\nruntime.goexit\n\t/snap/go/7736/src/runtime/asm_amd64.s:1371"]
BenchmarkTableScan-16                          	[2021/06/28 15:05:36.413 +08:00] [ERROR] [handle.go:683] ["[stats] dump stats delta fail"] [error="context canceled"] [errorVerbose="context canceled\ngithub.com/pingcap/errors.AddStack\n\t/home/genius/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.Trace\n\t/home/genius/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/juju_adaptor.go:15\ngithub.com/tikv/client-go/v2/locate.(*RegionRequestSender).onSendFail\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/locate/region_request.go:789\ngithub.com/tikv/client-go/v2/locate.(*RegionRequestSender).sendReqToRegion\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/locate/region_request.go:751\ngithub.com/tikv/client-go/v2/locate.(*RegionRequestSender).SendReqCtx\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/locate/region_request.go:529\ngithub.com/tikv/client-go/v2/tikv.(*ClientHelper).SendReqCtx\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/tikv/client_helper.go:83\ngithub.com/tikv/client-go/v2/tikv.(*KVSnapshot).get\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/tikv/snapshot.go:521\ngithub.com/tikv/client-go/v2/tikv.(*KVSnapshot).Get\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/tikv/snapshot.go:444\ngithub.com/pingcap/tidb/store/driver/txn.(*tikvSnapshot).Get\n\t/home/genius/project/src/github.com/pingcap/tidb/store/driver/txn/snapshot.go:45\ngithub.com/pingcap/tidb/executor.(*PointGetExecutor).get\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/point_get.go:405\ngithub.com/pingcap/tidb/executor.(*PointGetExecutor).Next\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/point_get.go:236\ngithub.com/pingcap/tidb/executor.Next\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/executor.go:286\ngithub.com/pingcap/tidb/executor.(*UpdateExec).updateRows\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/update.go:251\ngithub.com/pingcap/tidb/executor.(*UpdateExec).Next\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/update.go:219\ngithub.com/pingcap/tidb/executor.Next\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/executor.go:286\ngithub.com/pingcap/tidb/executor.(*ExecStmt).handleNoDelayExecutor\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/adapter.go:589\ngithub.com/pingcap/tidb/executor.(*ExecStmt).handleNoDelay\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/adapter.go:470\ngithub.com/pingcap/tidb/executor.(*ExecStmt).Exec\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/adapter.go:419\ngithub.com/pingcap/tidb/session.runStmt\n\t/home/genius/project/src/github.com/pingcap/tidb/session/session.go:1635\ngithub.com/pingcap/tidb/session.(*session).ExecuteStmt\n\t/home/genius/project/src/github.com/pingcap/tidb/session/session.go:1530\ngithub.com/pingcap/tidb/session.(*session).ExecuteInternal\n\t/home/genius/project/src/github.com/pingcap/tidb/session/session.go:1250\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).dumpTableStatCountToKV.func2\n\t/home/genius/project/src/github.com/pingcap/tidb/statistics/handle/update.go:473\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).dumpTableStatCountToKV\n\t/home/genius/project/src/github.com/pingcap/tidb/statistics/handle/update.go:477\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).DumpStatsDeltaToKV\n\t/home/genius/project/src/github.com/pingcap/tidb/statistics/handle/update.go:425\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).FlushStats\n\t/home/genius/project/src/github.com/pingcap/tidb/statistics/handle/handle.go:682\ngithub.com/pingcap/tidb/domain.(*Domain).updateStatsWorker\n\t/home/genius/project/src/github.com/pingcap/tidb/domain/domain.go:1279\nruntime.goexit\n\t/snap/go/7736/src/runtime/asm_amd64.s:1371"]
[2021/06/28 15:05:37.837 +08:00] [ERROR] [handle.go:683] ["[stats] dump stats delta fail"] [error="context canceled"] [errorVerbose="context canceled\ngithub.com/pingcap/errors.AddStack\n\t/home/genius/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.Trace\n\t/home/genius/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/juju_adaptor.go:15\ngithub.com/tikv/client-go/v2/locate.(*RegionRequestSender).onSendFail\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/locate/region_request.go:789\ngithub.com/tikv/client-go/v2/locate.(*RegionRequestSender).sendReqToRegion\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/locate/region_request.go:751\ngithub.com/tikv/client-go/v2/locate.(*RegionRequestSender).SendReqCtx\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/locate/region_request.go:529\ngithub.com/tikv/client-go/v2/tikv.(*ClientHelper).SendReqCtx\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/tikv/client_helper.go:83\ngithub.com/tikv/client-go/v2/tikv.(*KVSnapshot).get\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/tikv/snapshot.go:521\ngithub.com/tikv/client-go/v2/tikv.(*KVSnapshot).Get\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/tikv/snapshot.go:444\ngithub.com/pingcap/tidb/store/driver/txn.(*tikvSnapshot).Get\n\t/home/genius/project/src/github.com/pingcap/tidb/store/driver/txn/snapshot.go:45\ngithub.com/pingcap/tidb/executor.(*PointGetExecutor).get\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/point_get.go:405\ngithub.com/pingcap/tidb/executor.(*PointGetExecutor).Next\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/point_get.go:236\ngithub.com/pingcap/tidb/executor.Next\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/executor.go:286\ngithub.com/pingcap/tidb/executor.(*UpdateExec).updateRows\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/update.go:251\ngithub.com/pingcap/tidb/executor.(*UpdateExec).Next\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/update.go:219\ngithub.com/pingcap/tidb/executor.Next\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/executor.go:286\ngithub.com/pingcap/tidb/executor.(*ExecStmt).handleNoDelayExecutor\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/adapter.go:589\ngithub.com/pingcap/tidb/executor.(*ExecStmt).handleNoDelay\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/adapter.go:470\ngithub.com/pingcap/tidb/executor.(*ExecStmt).Exec\n\t/home/genius/project/src/github.com/pingcap/tidb/executor/adapter.go:419\ngithub.com/pingcap/tidb/session.runStmt\n\t/home/genius/project/src/github.com/pingcap/tidb/session/session.go:1635\ngithub.com/pingcap/tidb/session.(*session).ExecuteStmt\n\t/home/genius/project/src/github.com/pingcap/tidb/session/session.go:1530\ngithub.com/pingcap/tidb/session.(*session).ExecuteInternal\n\t/home/genius/project/src/github.com/pingcap/tidb/session/session.go:1250\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).dumpTableStatCountToKV.func2\n\t/home/genius/project/src/github.com/pingcap/tidb/statistics/handle/update.go:473\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).dumpTableStatCountToKV\n\t/home/genius/project/src/github.com/pingcap/tidb/statistics/handle/update.go:477\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).DumpStatsDeltaToKV\n\t/home/genius/project/src/github.com/pingcap/tidb/statistics/handle/update.go:425\ngithub.com/pingcap/tidb/statistics/handle.(*Handle).FlushStats\n\t/home/genius/project/src/github.com/pingcap/tidb/statistics/handle/handle.go:682\ngithub.com/pingcap/tidb/domain.(*Domain).updateStatsWorker\n\t/home/genius/project/src/github.com/pingcap/tidb/domain/domain.go:1279\nruntime.goexit\n\t/snap/go/7736/src/runtime/asm_amd64.s:1371"]
    6048	    204382 ns/op
BenchmarkExplainTableScan-16                   	[2021/06/28 15:05:38.218 +08:00] [ERROR] [domain.go:517] ["reload schema in loop failed"] [error="context canceled"] [errorVerbose="context canceled\ngithub.com/pingcap/errors.AddStack\n\t/home/genius/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.Trace\n\t/home/genius/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/juju_adaptor.go:15\ngithub.com/tikv/client-go/v2/locate.(*RegionRequestSender).onSendFail\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/locate/region_request.go:789\ngithub.com/tikv/client-go/v2/locate.(*RegionRequestSender).sendReqToRegion\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/locate/region_request.go:751\ngithub.com/tikv/client-go/v2/locate.(*RegionRequestSender).SendReqCtx\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/locate/region_request.go:529\ngithub.com/tikv/client-go/v2/tikv.(*ClientHelper).SendReqCtx\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/tikv/client_helper.go:83\ngithub.com/tikv/client-go/v2/tikv.(*KVSnapshot).get\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/tikv/snapshot.go:521\ngithub.com/tikv/client-go/v2/tikv.(*KVSnapshot).Get\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/tikv/snapshot.go:444\ngithub.com/pingcap/tidb/store/driver/txn.(*tikvSnapshot).Get\n\t/home/genius/project/src/github.com/pingcap/tidb/store/driver/txn/snapshot.go:45\ngithub.com/pingcap/tidb/structure.(*TxStructure).Get\n\t/home/genius/project/src/github.com/pingcap/tidb/structure/string.go:36\ngithub.com/pingcap/tidb/structure.(*TxStructure).GetInt64\n\t/home/genius/project/src/github.com/pingcap/tidb/structure/string.go:45\ngithub.com/pingcap/tidb/meta.(*Meta).GetSchemaVersion\n\t/home/genius/project/src/github.com/pingcap/tidb/meta/meta.go:260\ngithub.com/pingcap/tidb/domain.(*Domain).loadInfoSchema\n\t/home/genius/project/src/github.com/pingcap/tidb/domain/domain.go:104\ngithub.com/pingcap/tidb/domain.(*Domain).Reload\n\t/home/genius/project/src/github.com/pingcap/tidb/domain/domain.go:347\ngithub.com/pingcap/tidb/domain.(*Domain).loadSchemaInLoop\n\t/home/genius/project/src/github.com/pingcap/tidb/domain/domain.go:515\nruntime.goexit\n\t/snap/go/7736/src/runtime/asm_amd64.s:1371"]
[2021/06/28 15:05:40.266 +08:00] [ERROR] [domain.go:517] ["reload schema in loop failed"] [error="context canceled"] [errorVerbose="context canceled\ngithub.com/pingcap/errors.AddStack\n\t/home/genius/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.Trace\n\t/home/genius/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/juju_adaptor.go:15\ngithub.com/tikv/client-go/v2/locate.(*RegionRequestSender).onSendFail\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/locate/region_request.go:789\ngithub.com/tikv/client-go/v2/locate.(*RegionRequestSender).sendReqToRegion\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/locate/region_request.go:751\ngithub.com/tikv/client-go/v2/locate.(*RegionRequestSender).SendReqCtx\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/locate/region_request.go:529\ngithub.com/tikv/client-go/v2/tikv.(*ClientHelper).SendReqCtx\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/tikv/client_helper.go:83\ngithub.com/tikv/client-go/v2/tikv.(*KVSnapshot).get\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/tikv/snapshot.go:521\ngithub.com/tikv/client-go/v2/tikv.(*KVSnapshot).Get\n\t/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-20210621030356-e300b97684c7/tikv/snapshot.go:444\ngithub.com/pingcap/tidb/store/driver/txn.(*tikvSnapshot).Get\n\t/home/genius/project/src/github.com/pingcap/tidb/store/driver/txn/snapshot.go:45\ngithub.com/pingcap/tidb/structure.(*TxStructure).Get\n\t/home/genius/project/src/github.com/pingcap/tidb/structure/string.go:36\ngithub.com/pingcap/tidb/structure.(*TxStructure).GetInt64\n\t/home/genius/project/src/github.com/pingcap/tidb/structure/string.go:45\ngithub.com/pingcap/tidb/meta.(*Meta).GetSchemaVersion\n\t/home/genius/project/src/github.com/pingcap/tidb/meta/meta.go:260\ngithub.com/pingcap/tidb/domain.(*Domain).loadInfoSchema\n\t/home/genius/project/src/github.com/pingcap/tidb/domain/domain.go:104\ngithub.com/pingcap/tidb/domain.(*Domain).Reload\n\t/home/genius/project/src/github.com/pingcap/tidb/domain/domain.go:347\ngithub.com/pingcap/tidb/domain.(*Domain).loadSchemaInLoop\n\t/home/genius/project/src/github.com/pingcap/tidb/domain/domain.go:515\nruntime.goexit\n\t/snap/go/7736/src/runtime/asm_amd64.s:1371"]
   19360	     54308 ns/op
```

So many annoying log... I can't read the result!

Problem Summary:

### What is changed and how it works?

I try to run some regression test https://github.com/pingcap/tidb/issues/25754, but I find some of the code is broken.

What's Changed:

Avoid unnecessary log.

How it Works:

1. Close `domain` before `store`, because if we close store first, the domain is still using the store, thus there will be error output.
2. Move `sysSessionPool.Close()` after statistic handle close, because the later is still using the session pool on exit
3. Disable the slow query log in bench test to avoid the slow log output

### Check List 

Tests 

- No code

### Release note 

- No release note

### CODE REVIEW

File: session/bench_test.go [hunk 1]
```diff
@@ -36,6 +37,10 @@ var smallCount = 100
 var bigCount = 10000
 
 func prepareBenchSession() (Session, *domain.Domain, kv.Storage) {
+	config.UpdateGlobal(func(cfg *config.Config) {
```
Comment (xhebox) [inline session/bench_test.go:40]: Will session variable `TiDBEnableSlowLog` take effects? I think it is better than global config, if it works.

**Review by xhebox — COMMENTED** [review]
Rest LGTM

**Review by tangenta — APPROVED** [review]

File: session/bench_test.go (hunk 1 above)
Comment (tiancaiamao) [reply session/bench_test.go:40]: I'm not sure is the CI stable currently, or is it affected by this change, let me re-run it.

**Review by xhebox — APPROVED** [review]

**Comment by tiancaiamao** [comment]
```
[2021-06-29T09:43:06.864Z] ==================
[2021-06-29T09:43:06.864Z] WARNING: DATA RACE
[2021-06-29T09:43:06.864Z] Read at 0x00c027332d78 by goroutine 94:
[2021-06-29T09:43:06.864Z]   internal/race.Read()
[2021-06-29T09:43:06.864Z]       /usr/local/go/src/internal/race/race.go:37 +0x206
[2021-06-29T09:43:06.864Z]   sync.(*WaitGroup).Add()
[2021-06-29T09:43:06.864Z]       /usr/local/go/src/sync/waitgroup.go:71 +0x219
[2021-06-29T09:43:06.864Z]   github.com/tikv/client-go/v2/tikv.(*twoPhaseCommitter).doActionOnGroupMutations()
[2021-06-29T09:43:06.864Z]       /nfs/cache/mod/github.com/tikv/client-go/v2@v2.0.0-alpha.0.20210628064110-666340265aac/tikv/2pc.go:680 +0x7a8
[2021-06-29T09:43:06.864Z]   github.com/tikv/client-go/v2/tikv.(*twoPhaseCommitter).doActionOnMutations()
[2021-06-29T09:43:06.864Z]       /nfs/cache/mod/github.com/tikv/client-go/v2@v2.0.0-alpha.0.20210628064110-666340265aac/tikv/2pc.go:532 +0x17a
[2021-06-29T09:43:06.864Z]   github.com/tikv/client-go/v2/tikv.(*twoPhaseCommitter).commitMutations()
[2021-06-29T09:43:06.864Z]       /nfs/cache/mod/github.com/tikv/client-go/v2@v2.0.0-alpha.0.20210628064110-666340265aac/tikv/commit.go:208 +0x109
[2021-06-29T09:43:06.864Z]   github.com/tikv/client-go/v2/tikv.(*twoPhaseCommitter).commitTxn()
[2021-06-29T09:43:06.864Z]       /nfs/cache/mod/github.com/tikv/client-go/v2@v2.0.0-alpha.0.20210628064110-666340265aac/tikv/2pc.go:1311 +0x317
[2021-06-29T09:43:06.864Z]   github.com/tikv/client-go/v2/tikv.(*twoPhaseCommitter).execute()
[2021-06-29T09:43:06.864Z]       /nfs/cache/mod/github.com/tikv/client-go/v2@v2.0.0-alpha.0.20210628064110-666340265aac/tikv/2pc.go:1302 +0x1a5c
[2021-06-29T09:43:06.864Z]   github.com/tikv/client-go/v2/tikv.(*KVTxn).Commit()
[2021-06-29T09:43:06.864Z]       /nfs/cache/mod/github.com/tikv/client-go/v2@v2.0.0-alpha.0.20210628064110-666340265aac/tikv/txn.go:393 +0x572
[2021-06-29T09:43:06.864Z]   github.com/pingcap/tidb/store/driver/txn.(*tikvTxn).Commit()
[2021-06-29T09:43:06.864Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/store/driver/txn/txn_driver.go:66 +0x64
[2021-06-29T09:43:06.864Z]   github.com/pingcap/tidb/kv.RunInNewTxn()
[2021-06-29T09:43:06.864Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/kv/txn.go:60 +0x687
[2021-06-29T09:43:06.864Z]   github.com/pingcap/tidb/ddl.(*delRange).doTask()
[2021-06-29T09:43:06.864Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/delete_range.go:198 +0x252
[2021-06-29T09:43:06.864Z]   github.com/pingcap/tidb/ddl.(*delRange).doDelRangeWork()
[2021-06-29T09:43:06.864Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/delete_range.go:184 +0x568
[2021-06-29T09:43:06.864Z]   github.com/pingcap/tidb/ddl.(*delRange).startEmulator()
[2021-06-29T09:43:06.864Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/ddl/delete_range.go:148 +0x204
[2021-06-29T09:43:06.864Z] 
[2021-06-29T09:43:06.864Z] Previous write at 0x00c027332d78 by goroutine 444:
[2021-06-29T09:43:06.864Z]   internal/race.Write()
[2021-06-29T09:43:06.864Z]       /usr/local/go/src/internal/race/race.go:41 +0x125
[2021-06-29T09:43:06.864Z]   sync.(*WaitGroup).Wait()
[2021-06-29T09:43:06.864Z]       /usr/local/go/src/sync/waitgroup.go:128 +0x126
[2021-06-29T09:43:06.865Z]   github.com/tikv/client-go/v2/tikv.(*KVStore).Close()
[2021-06-29T09:43:06.865Z]       /nfs/cache/mod/github.com/tikv/client-go/v2@v2.0.0-alpha.0.20210628064110-666340265aac/tikv/kv.go:296 +0x6b
[2021-06-29T09:43:06.865Z]   github.com/pingcap/tidb/store/mockstore/mockstorage.(*mockStorage).Close()
[2021-06-29T09:43:06.865Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/store/mockstore/mockstorage/storage.go:127 +0x75
[2021-06-29T09:43:06.865Z]   github.com/pingcap/tidb/executor_test.(*testRecoverTable).TearDownSuite()
[2021-06-29T09:43:06.865Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/executor/executor_test.go:5736 +0x50
[2021-06-29T09:43:06.865Z]   runtime.call16()
[2021-06-29T09:43:06.865Z]       /usr/local/go/src/runtime/asm_amd64.s:550 +0x3d
[2021-06-29T09:43:06.865Z]   reflect.Value.Call()
[2021-06-29T09:43:06.865Z]       /usr/local/go/src/reflect/value.go:337 +0xd8
[2021-06-29T09:43:06.865Z]   github.com/pingcap/check.(*suiteRunner).runFixture.func1()
```

Data race when some background job is using store and the other goroutine close the store.
Not caused by this change.
I'll file a issue in the client-go repo.

**Comment by tiancaiamao** [comment]
/merge cancel

DATA RACE again, I need to investigate ...

File: executor/index_lookup_hash_join.go [hunk 2]
```diff
@@ -598,10 +598,10 @@ func (iw *indexHashJoinInnerWorker) handleTask(ctx context.Context, task *indexH
 	// TODO(XuHuaiyu): we may always use the smaller side to build the hashtable.
 	go util.WithRecovery(func() { iw.buildHashTableForOuterResult(ctx, task, h) }, iw.handleHashJoinInnerWorkerPanic)
 	err := iw.fetchInnerResults(ctx, task.lookUpJoinTask)
+	iw.wg.Wait()
```
Comment (tiancaiamao) [inline executor/index_lookup_hash_join.go:601]: This change fix https://github.com/pingcap/tidb/issues/25830

File: executor/executor_test.go [hunk 3]
```diff
@@ -5733,8 +5733,8 @@ func (s *testRecoverTable) SetUpSuite(c *C) {
 }
 
 func (s *testRecoverTable) TearDownSuite(c *C) {
-	s.store.Close()
 	s.dom.Close()
+	s.store.Close()
```
Comment (tiancaiamao) [inline executor/executor_test.go:5737]: This change fix https://github.com/pingcap/tidb/issues/25831

**Pull request merged**